# Ejercicios de probabilidad

In [1]:
import random
import numpy as np

## Ejercicio 1 

Dos dados se lanzan una vez y se observa el total obtenido. Usa una simulación para encontrar la probabilidad estimada de que la puntuación sea mayor a 7 o un número par. Una simulación es una repetición del mismo experimento multiples veces para observar su comportamiento:

- Ejecuta el experimento 1000 veces (lanza 2 dados 1000 veces, y suma el número de ambos dados).
- Lleva ua cuenta de los números y las veces que la suma fue mayor a 7 o un número par.
- Divide el número del paso 2 entre el número de iteraciones(1000).

In [5]:
# Número de experimentos (lanzamientos)
num_experiments = 1000

# Simulamos los lanzamientos de dos dados (valores entre 1 y 6)
dados_1 = np.random.randint(1, 7, size=num_experiments)
dados_2 = np.random.randint(1, 7, size=num_experiments)

# Sumamos los valores de ambos dados
suma = dados_1 + dados_2

# Verificamos la condición: suma > 7 o suma par
condicion = (suma > 7) | (suma % 2 == 0)

# Contamos cuántas veces se cumple la condición
conteo = np.sum(condicion)

# Calculamos la probabilidad
probabilidad = conteo / num_experiments

# Mostramos el resultado
round(probabilidad, 2)


np.float64(0.67)

# Conclusion:
> La probabilidad de que la suma sea mayor a 7 o par es 0,67

## Ejercicio 2

Una caja contiene 10 bolas blancas, 20 bolas rojas y 30 bolas verdes. Si tomamos 5 bolas de la caja reemplazándolas (tomando una bola, anotando el color y luego regresandola a la caja). Queremos saber la probabilidad de:

1. Tomar 3 blancas y 2 rojas.
2. Toas todas del mismo color.

Ejecuta el experimento 1000 veces y calcula las probabilidades que mencionadas.

In [6]:
ball_box = {}

# Crea la cajas con las bolas
for i in range(60):
    if i < 10:
        ball_box[i] = "White"
    elif (i > 9) and (i < 30):
        ball_box[i] = "Red"
    else:
        ball_box[i] = "Green"

# Número de experimentos
num_experiments = 1000

# Creamos la caja de bolas como un array de colores
# 10 Blancas, 20 Rojas, 30 Verdes
ball_box = np.array(['White'] * 10 + ['Red'] * 20 + ['Green'] * 30)

# Realizamos todas las extracciones de una vez: (1000 x 5)
extracciones = np.random.choice(ball_box, size=(num_experiments, 5), replace=True)

# Contamos cuántas bolas blancas y rojas hay por extracción
blancas = np.sum(extracciones == 'White', axis=1)
rojas = np.sum(extracciones == 'Red', axis=1)

# Condición: 3 blancas y 2 rojas
cond_3_blancas_2_rojas = (blancas == 3) & (rojas == 2)

# Condición: todas del mismo color (las 5 iguales)
cond_todas_mismo_color = np.all(extracciones == extracciones[:, [0]], axis=1)

# Calculamos probabilidades
prob_3_blancas_2_rojas = np.mean(cond_3_blancas_2_rojas)
prob_todas_mismo_color = np.mean(cond_todas_mismo_color)

# Imprimimos los resultados redondeados
round(prob_3_blancas_2_rojas, 3), round(prob_todas_mismo_color, 3)




(np.float64(0.001), np.float64(0.034))

# Conclusion:
>La probabilidad de tomar 3 blancas y 2 rojas: 0.001. La probabilidad de tomar todas del mismo color: 0.034